packages/commerce-sdk-react/src/hooks/ShopperBaskets/query.ts

```diff
@@ -189,13 +201,16 @@ export const useTaxesFromBasket = (
     type Data = DataType<Client['getTaxesFromBasket']>
     const {shopperBaskets: client} = useCommerceApi()
     const methodName = 'getTaxesFromBasket'
-    const requiredParameters = ['organizationId', 'basketId', 'siteId'] as const
+    const requiredParameters = ShopperBaskets.paramKeys[`${methodName}Required`]
 
     // Parameters can be set in `apiOptions` or `client.clientConfig`;
     // we must merge them in order to generate the correct query key.
     const netOptions = omitNullableParameters(mergeOptions(client, apiOptions))
     // get param keys for the api from netOptions
-    const paramKeys = [...paramKeysMap[methodName], ...getCustomKeys(netOptions.parameters)]
+    const paramKeys = [
```

Comment:
I am wondering if we can create a util that returns the keys including custom ones. 🤔 . With this we can optimise a few places (here and queryKeyHelper.queryKeys) 
```
const paramKeys = getParamKeys(medthodName, params)


// util.js
const getParamKeys = (medthodName, params)  => {
   // add logic to get the right shopperClass here
   return [
     ....Shopper*[medthodName],
     ...getCustomKeys(params)
   ]

}

```

Reply:
Good idea @alexvuong 👍

Reply:
@alexvuong please take a look at the latest changes. Inspired by your idea, I took it further and created a util method that picks the valid parameters. That's what the param keys are originally used for: to pick the valid parameters.